sktime/annotation/tests/test_all_annotators.py

```diff
@@ -20,8 +21,16 @@ def test_output_type(Estimator):
 
     estimator = Estimator.create_test_instance()
 
-    arg = make_annotation_problem(n_timepoints=50)
-    estimator.fit(arg)
-    arg = make_annotation_problem(n_timepoints=10)
-    y_pred = estimator.predict(arg)
-    assert isinstance(y_pred, (pd.Series, np.ndarray))
+    if estimator.get_tag("distribution_type") == "Poisson":
+        arg = piecewise_poisson(lambdas=[1, 2, 3], lengths=[2, 4, 8], random_state=42)
+        estimator.fit(arg)
+        arg = piecewise_poisson(lambdas=[1, 3, 6], lengths=[2, 4, 8], random_state=42)
+        y_pred = estimator.predict(arg)
+        assert isinstance(y_pred, (pd.Series, np.ndarray))
+
+    else:
+        arg = make_annotation_problem(n_timepoints=50)
```

Comment:
Excellent!  Thank you for implementing the tag system, I think this is looking much better! :) 

However I do have one request (not on the logic itself but rather where we put it. To me - it makes more sense to add the distribution dependent logic within `make_annotation_problem` rather than in this particular test.  The thought being - in the future we may add more tests for the annotator classes in general (this part of the library is still pretty new) and we would want those tests to be able to make use of the `make_annotation_problem` function. Rather than have to check in each test what distribution is needed, it makes more sense to me to pass this logic off to `make_annotation_problem`. 

It should be a simple matter of changing `make_annotation_problem` to (optionally) take in the estimator as well.  Then if an estimator is passed it can run some checks on the tags and tailor the problem to fit the needs of that particular estimator. 

Again, thank you for your efforts team, this is looking good! :)

Reply:
Thank you for the review!

We have made the changes to make_annotation_problem as well as make_series in order to accommodate for the change in annotator class. We have left an estimator_type argument in the make_annotation_problem for future instances.

Reply:
Hey!

Thank you guys for updating this in response to my suggestions - it is really looking much better!

I think we can push this just a bit farther and separate the new logic here a bit better from the existing code base.  To motivate what I am thinking/why I am asking for it a bit more:

- in sktime/object oriented programming in general we want to write code that respects "separation of concerns".  In this instance - I am a bit concerned that by putting all this logic about the distribution type in `make_series` we are actually adding some functionality and logic into that class when it doesn't strictly need to be there!  Again, this isn't breaking the code - but the added complexity can make maintenance and debugging a bit more difficult.
- on the other hand you guys just wrote a nice function for generating exactly the type of data that would work well for `PoissonHMM` (or any other estimator that needed Poisson like data!). So the logic we need already exists in sktime!  

Personally I kind of think of the functionality of `make_annotation_problem` to be acting as a delegator.  So far it has only delegated to `make_series`, but we can change it to instead optionally delegate the role of making an example problem to `piecewise_poisson` instead! 

Let me know what you think on the above!  Again, not intended as a huge change request, but I do think it would make things cleaner here, and also is a great opportunity to use that new function you guys wrote as well! :) 

- Mirae

Reply:
Nice!  Looks much better now!  Thank you guys! :)